### PR TITLE
Feature/cvsb 7949

### PR DESCRIPTION
**Pull request by bfloreaDVSA**
added tests for 7949

### CODE REVIEW

**Review by jgimber-deloitte — APPROVED** [review]

**Review by dpanzaru-deloitte — CHANGES_REQUESTED** [review]

File: src/main/java/steps/TestResultsSteps.java [hunk 1]
```diff
@@ -328,10 +329,17 @@ private void validateLocation(TestResults testResults) {
     @Step
     public void validatePostErrorData(String field, String errorMessage) {
         response.then().body("size()", is(1));
-        response.then().body("errors.size()", is(1));
+        response.then().body("errors.size()", greaterThanOrEqualTo(1));
```
[the remark this answers is not in the document]
Comment (bfloreaDVSA) [reply src/main/java/steps/TestResultsSteps.java]: when you run this test, on the first run you'll have 1 error, from the second one forth, each run will add another, so you can only be sure of one

File: src/main/java/steps/TestResultsSteps.java [hunk 2]
```diff
@@ -411,4 +419,108 @@ public void checkNextTestNumberIsValid(){
         String testNumber = response.jsonPath().getString("[0].testTypes[0].testNumber");
         assertThat(testNumber.equals(nextTestNumber)).isTrue();
     }
+
+    @Step
+    public void validateVehicleFieldValue(String key, String value) {
+        response.then().log().all();
```
Comment (jgimber-deloitte) [inline src/main/java/steps/TestResultsSteps.java]: This won't show the bearer token in the log, right?

**Review by jgimber-deloitte — CHANGES_REQUESTED** [review]
Just need to verify that no bearer token info isn't put into the logs...

**Review by jgimber-deloitte — DISMISSED** [review]

File: src/main/java/steps/TestResultsSteps.java (hunk 2 above)
Comment (bfloreaDVSA) [reply src/main/java/steps/TestResultsSteps.java]: nope, it's on response

File: src/test/java/testresults/GetTestResults.java [hunk 3]
```diff
@@ -123,4 +118,43 @@ public void testResultsWithStatusInvalidAndNotExistingVin() {
...
+        testResultsSteps.validateVehicleFieldExists("numberOfSeats");
+        testResultsSteps.validateVehicleFieldExists("odometerReading");
+        testResultsSteps.validateVehicleFieldExists("odometerReadingUnits");
+        testResultsSteps.validateVehicleFieldExists("vehicleSize");
+        testResultsSteps.validateTestFieldExists("numberOfSeatbeltsFitted");
+        testResultsSteps.validateTestFieldExists("lastSeatbeltInstallationCheckDate");
+        testResultsSteps.validateTestFieldExists("seatbeltInstallationCheckDate");
+    }
+
+@Title("CVSB-6805 - API Consumer retrieve the Test results for the input Vin (HGV)")
```
[the remark this answers is not in the document]
Comment (bfloreaDVSA) [reply src/test/java/testresults/GetTestResults.java]: done

File: src/test/java/testresults/GetTestResults.java [hunk 4]
```diff
@@ -123,4 +118,43 @@ public void testResultsWithStatusInvalidAndNotExistingVin() {
...
+        testResultsSteps.getTestResults("P012301230123");
+        testResultsSteps.statusCodeShouldBe(200);
+        testResultsSteps.validateVehicleFieldValue("vin", "P012301230123");
+        testResultsSteps.validateVehicleFieldValue("vehicleType", "hgv");
+        testResultsSteps.validateVehicleFieldExists("vrm");
+        testResultsSteps.validateVehicleFieldExists("odometerReading");
+        testResultsSteps.validateVehicleFieldExists("odometerReadingUnits");
+    }
+
+@Title("CVSB-6805 - API Consumer retrieve the Test results for the input Vin (TRL)")
```
[the remark this answers is not in the document]
Comment (bfloreaDVSA) [reply src/test/java/testresults/GetTestResults.java]: done

**Review by jgimber-deloitte — DISMISSED** [review]

**Review by dpanzaru-deloitte — APPROVED** [review]

**Review by jgimber-deloitte — APPROVED** [review]

**Pull request merged**